integrationtest/osgi/karaf-it/src/test/java/org/hibernate/search/test/integration/osgi/HibernateSearchWithKarafIT.java

```diff
@@ -138,6 +148,24 @@ public void tearDown() {
 		bundleContext.ungetService( serviceReference );
 	}
 
+	/**
+	 * Starts all bundles to make sure there are no "uses constraint violations" caused by packages exported from the
+	 * bundles contributed by the HSEARCH Karaf feature.
+	 */
+	@Test
+	@TestForIssue(jiraKey="HSEARCH-1774")
+	public void testStartingAllBundles() throws Exception {
+		for ( Bundle bundle : bundleContext.getBundles() ) {
+			if ( !isFragmentBundle( bundle ) ) {
+				bundle.start();
+			}
+		}
+	}
+
+	private boolean isFragmentBundle(Bundle bundle) {
```

Comment:
> It's a bundle which does not live on its own but attaches to another bundle (the "fragment host")

Ahh, something learned here. I did not know about this. Thanks.

> Fragments may actually be useful for us if e.g. engine is a fragment host and other HS modules are fragments of that. 

Interesting indeed. This could have been useful for engine + orm. Well, on the other hand it is nice to have a "clean" package separation and no impl dependencies.